test/topic_SUITE.erl

```diff
@@ -95,3 +95,17 @@ index_bug(Config) ->
 
     ?assertEqual(199, HWM4),
     ?assertEqual(190, length(Reply4)).
+
+limit(Config) ->
```

Comment:
I'd be interested in a test that also does something like max bytes of 1. I'd worry that the client might break in a case that the max bytes is < the size of one record.